### ISSUE TITLE
Numpy version incompatibility when installing llama_cpp_python for GPU acceleration

### ISSUE THREAD

**Issue by mdevino** [instructlab/instructlab#1666]
**Describe the bug**

I was following the steps on gpu-acceleration.md and, when executing [this line](https://github.com/instructlab/instructlab/blob/main/docs/gpu-acceleration.md#:~:text=cache%20remove%20llama_cpp_python-,pip%20install%20%2D%2Dforce%2Dreinstall%20llama_cpp_python%3D%3D0.2.79%20%2DC%20cmake.args%3D%22%2DDLLAMA_CUDA%3Don%22,-%23%20Re%2Dinstall), I received the output below:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
transformers 4.42.3 requires numpy<2.0,>=1.17, but you have numpy 2.0.0 which is incompatible.
trl 0.9.6 requires numpy<2.0.0,>=1.18.2, but you have numpy 2.0.0 which is incompatible.
instructlab 0.17.1.dev380 requires numpy<2.0.0,>=1.26.4; python_version != "3.10", but you have numpy 2.0.0 which is incompatible.
accelerate 0.32.1 requires numpy<2.0.0,>=1.17, but you have numpy 2.0.0 which is incompatible.
instructlab-training 0.1.0 requires numpy<2.0.0,>=1.26.4; python_version != "3.10", but you have numpy 2.0.0 which is incompatible.
```

**To Reproduce**
1. Clone [InstructLab](https://github.com/instructlab/instructlab)
2. Execute the steps in [Nvidia/CUDA section](https://github.com/instructlab/instructlab/blob/main/docs/gpu-acceleration.md#nvidiacuda)
3. When executing the command `pip install --force-reinstall llama_cpp_python==0.2.79 -C cmake.args="-DLLAMA_CUDA=on"
`, the output above would be shown.

**Expected behavior**

I would expect to successfully run `ilab model chat` with GPU acceperation.

**Screenshots**

Here's a screenshot of the aforementioned output.

![image](https://github.com/instructlab/instructlab/assets/19861348/00b3fb7b-e0aa-4810-8b50-b2eab6569efe)

**Device Info (please complete the following information):**
 - Hardware Specs: CPU: Intel i9-10900k, GPU: RTX 3080
 - OS Version: ArchLinux  on WSL2 (output of `uname -srvm`: `Linux 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64`)
 - Python Version: Python 3.11.9
 - InstructLab Version: ilab, version 0.17.1.dev380

**Additional context**

**Comment by mdevino** [comment]
I've just created #1667 with what worked for me. Please let me know if further changes are needed (or this issue is not relevant fix, and is too "user-specific").

**Comment by mdevino** [comment]
Here are some screenshots of it working for me after the changes.

Prompt:
![image](https://github.com/instructlab/instructlab/assets/19861348/45f07ba5-68d2-47e9-b5f0-f294afda16f8)

GPU usage:
![image](https://github.com/instructlab/instructlab/assets/19861348/34cd44d1-1245-498b-8b9e-3659e84cfb58)

**Comment by nathan-weinberg** [comment]
Fixed by #1667
